src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -87,6 +105,177 @@
 ApplicationOwnedObject = TypedDict("ApplicationOwnedObject", {"name": str, "type": str})
 
 
+class EventSharingHandler:
+    """
+    Handles the logic around event sharing for applications.
+    This class is used to determine whether event sharing should be authorized or not, and which events should be shared.
+    """
+
+    def _to_events_definitions(
+        self, all_events: List[str], mandatory_events: List[str]
+    ) -> List[Dict[str, str]]:
+        return [
+            {
+                "name": f"SNOWFLAKE${event}",
+                "type": event,
+                "sharing": "MANDATORY" if event in mandatory_events else "OPTIONAL",
+                "status": "DISABLED",
+            }
+            for event in all_events
+        ]
+
+    def __init__(
+        self,
+        telemetry_definition: Optional[EventSharingTelemetry],
+        deploy_root_path: Path,
+        install_method: SameAccountInstallMethod,
+        console: AbstractConsole,
+    ):
+        self._is_dev_mode = install_method.is_dev_mode
+        connection = get_sql_executor()._conn  # noqa: SLF001
```

Comment:
I just saw that we make conn.execute_string calls outside of the sql_executor (in connection/utils.py). Seems like not a great idea to have bypassed the sql_executor abstraction altogether. I know it _might_ be outside the scope of this PR to fix that all, but I agree with Bruno that we shouldn't use it here at least.

Reply:
We decided to revert this since the facade is NA specific, and utils is common code. We will move this to the Facade once it's used globally (not just for NA)